### PR TITLE
Fix builds

### PR DESCRIPTION
**Pull request by crtrott**
This fixes a bunch of warnings and build issues in particular in C++17/14

### CODE REVIEW

**Comment by crtrott** [comment]
This passed all the stuff in the config scripts I pushed up.

File: tests/test_submdspan_static_slice.cpp [hunk 1]
```diff
@@ -567,7 +567,7 @@ TEST(TestMdspan, SubmdspanStaticSlice_Left_idd_FullTuple) {
   // in order for the output's extents type to be able to encode
   // the compile-time slice information.
   static_assert(std::is_convertible<std::tuple<IC<std::size_t, 1>, IC<std::size_t, 3>>,
-                                    std::tuple<std::size_t, std::size_t>>::value);
+                                    std::tuple<std::size_t, std::size_t>>::value, "convertability of tuples of integral constant to size_t is false");
```
Comment (mhoemmen) [inline tests/test_submdspan_static_slice.cpp]: Oh oof, I missed that one! : - )  Thanks!!!

**Review by mhoemmen — APPROVED** [review]

File: tests/test_submdspan_static_slice.cpp (hunk 1 above)
Comment (crtrott) [reply tests/test_submdspan_static_slice.cpp]: its allowed in C++17

Comment (mhoemmen) [reply tests/test_submdspan_static_slice.cpp]: I had meant mdspan to be back-portable to C++14, but forgot this particular change.  Thanks for fixing it!

Comment (dalg24) [reply tests/test_submdspan_static_slice.cpp]: typo convertability -> convertibility

File: benchmarks/stencil/cuda/stencil_3d_cuda.cu [hunk 2]
```diff
@@ -103,13 +103,13 @@ inline void cuda_internal_safe_call(cudaError e, const char* name,
 
 dim3 get_bench_thread_block(size_t y,size_t z) {
   cudaDeviceProp cudaProp;
-  int dim_z = 1;
+  size_t dim_z = 1;
```
Comment (dalg24) [inline benchmarks/stencil/cuda/stencil_3d_cuda.cu:106]: why did you change?

Comment (crtrott) [reply benchmarks/stencil/cuda/stencil_3d_cuda.cu:106]: Warning undigned signed in the while loop

Comment (dalg24) [reply benchmarks/stencil/cuda/stencil_3d_cuda.cu:106]: cast `y` and `z`  to int instead when you compare

**Review by dalg24 — APPROVED** [review]
